### PR TITLE
Fix #2964: TreeSelect add expandedKeys property

### LINKED ISSUE #2964: TreeSelect: Add collapseAll and expandAll methods
### Describe the feature you would like to see added

I need a method to collapse all nodes in TreeSelect component.
This is because in my case, when I retrieve new data from API, the changed elements are re-rendered and nodes expand by themselves. 

### Is your feature request related to a problem?

The problem is that when the list fetches new data depending on what the user selects, all nodes expand. 
This is annoying because when the list is too long the user loses sight of what he was interested in. 

### Describe the solution you'd like

I guess it would be easiest if there was a callback function to collapse all nodes. Unless you have a better solution.

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by arkadiusztw**
Solves https://github.com/primefaces/primereact/issues/2964 for TreeSelect

### CODE REVIEW

File: components/lib/treeselect/treeselect.d.ts [hunk 1]
```diff
@@ -87,6 +94,7 @@ export interface TreeSelectProps extends Omit<React.DetailedHTMLProps<React.Inpu
     panelClassName?: string;
     appendTo?: TreeSelectAppendToType;
     emptyMessage?: string;
+    expandedKeys?: TreeSelectExpandedKeysType;
```
Comment (melloware) [inline components/lib/treeselect/treeselect.d.ts:97]: Please update `\api-generator\components\treeselect.js` with the new `expandedKeys` property as well.

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by melloware** [comment]
Also looks like you are getting a TypeScript failure: Error: components/lib/treeselect/treeselect.d.ts(60,45): error TS2709: Cannot use namespace 'TreeEventNodeParams' as a type.

Just run `npm run type-check` locally to compile your TS before pushing.

**Comment by melloware** [comment]
OK I am rerunning the build with your latest changes

**Review by melloware — APPROVED** [review]

File: pages/treeselect/index.js [hunk 2]
```diff
@@ -37,19 +59,76 @@ const TreeSelectDemo = () => {
...
+                        expandedKeys={expandedKeys}
+                        options={nodes}
+                        onToggle={(e) => setExpandedKeys(e.value)}
+                        onChange={(e) => setSelectedNodeKey3(e.value)}
+                        placeholder="Select Item"
+                    ></TreeSelect>
+
+                    <h5>Programmatic Control</h5>
+                    <div className="mb-4">
+                        <Button
```
Comment (melloware) [inline pages/treeselect/index.js:118]: Shoudln';t for this Programmatic Control Demo we only use 1 TreeSelect and use a new TreeSelect. I know Mert likes exact examples and not updating the previous examples?

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by melloware** [comment]
also if you make a comment on the original ticket it will let me assign you to the ticket.

**Review by melloware — APPROVED** [review]

**Comment by mertsincan** [comment]
Thanks a lot for your contribution! But, all the formatting in the files seems to have changed. Could you please update your files? After these changes, I or @melloware'll merge this nice feature.

**Comment by arkadiusztw** [comment]
Do you mean to use prettier on the file components/doc/treeselect/index.js? Or is it about something else?
Please clarify

**Comment by melloware** [comment]
No he means on your index.JS it looks like you formatted the examples. Just make the smallest necessary changes for review is all Mert is saying.

File: components/doc/treeselect/index.js [hunk 3]
```diff
@@ -75,40 +162,123 @@ const TreeSelectDemo = () => {
...
-                <h5>Single</h5>
-                <TreeSelect value={selectedNodeKey1} options={nodes} onChange={(e) => setSelectedNodeKey1(e.value)} placeholder="Select Item"></TreeSelect>
+        <div className="card">
+                    <h5>Single</h5>
+                    <TreeSelect
+                        value={selectedNodeKey1}
+                        options={nodes}
+                        onChange={(e) => setSelectedNodeKey1(e.value)}
+                        placeholder="Select Item"
+                    ></TreeSelect>
```
Comment (melloware) [inline components/doc/treeselect/index.js:205]: I think this is what Mert means leave these example flattened not wrapped

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/doc/treeselect/index.js (hunk 3 above)
Comment (melloware) [reply components/doc/treeselect/index.js:205]: I would just grab the latest index.js and just reapply your new example only.

**Comment by melloware** [comment]
@arkadiusztw any chance you can fix index.js?

**Comment by arkadiusztw** [comment]
> @arkadiusztw any chance you can fix index.js?

yes, I just need some time

**Comment by melloware** [comment]
Since we updated to Prettier I took care of this for you!  Thanks for your PR!

**Pull request closed without merging**